File: step/OC/oneOC/View/BoardCell.m
//
//  BoardCell.m
//  oneOC
//
//  Created by Jz D on 2020/4/6.
//  Copyright © 2020 Jz D. All rights reserved.
//

#import "BoardCell.h"



@interface BoardCell()



@end





@implementation BoardCell

#pragma mark -
#pragma mark Initialization
- (id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
 
    if (self) {
        // Cell Type
        self.cellType = BoardCellTypeEmpty;
    }
 
    return self;
}
 
#pragma mark -
#pragma mark Setters & Getters
- (void)setCellType: (BoardCellType)cellType {
    if (_cellType != cellType) {
        _cellType = cellType;
 
        // Update View
        [self updateView];
    }
}
 
#pragma mark -
#pragma mark Helper Methods
- (void)updateView {
    // Background Color
    
    if (self.cellType == BoardCellTypeMine) {
        self.backgroundColor = [UIColor yellowColor];
    }
    else if (self.cellType == BoardCellTypeYours){
        self.backgroundColor = [UIColor redColor];
    }
    else{
        self.backgroundColor = [UIColor whiteColor];
    }
}
 



@end

File: p/socketD/socketD/View/BoardCell.swift
//
//  BoardCell.swift
//  socketD
//
//  Created by Jz D on 2020/4/21.
//  Copyright © 2020 Jz D. All rights reserved.
//

import UIKit


enum BoardCellType: Int{
    case empty = -1, mine = 0, yours = 1
}




class BoardCell: UIView {

    var _cellType = BoardCellType.empty
    
    
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
    }
    
    required init?(coder: NSCoder) {
        fatalError()
    }
    
    var cellType: BoardCellType{
        get{
            return _cellType
        }
        set{
            if _cellType != newValue{
                _cellType = newValue
                updateView()
            }
        }
    }
    
    
        
    func updateView(){
        // Background Color
        switch cellType {
        case .mine:
            backgroundColor = UIColor.yellow
        case .yours:
            backgroundColor = UIColor.red
        default :
            //  .empty
            backgroundColor = UIColor.white
        }
    }
     

    

}
